pkg/referrerstore/oras/oras.go

```diff
@@ -373,7 +377,7 @@ func (store *orasStore) GetSubjectDescriptor(ctx context.Context, subjectReferen
 	return &ocispecs.SubjectDescriptor{Descriptor: desc}, nil
 }
 
-func (store *orasStore) createRepository(ctx context.Context, targetRef common.Reference) (*remote.Repository, time.Time, error) {
+func createDefaultRepository(ctx context.Context, store *orasStore, targetRef common.Reference) (registry.Repository, time.Time, error) {
```

Comment:
Just curious why we remove the receiver? `store.createRepository(ctx, store, subjectReference)` looks a bit strange to me that we pass in a `store` to its method.

Reply:
yeah I agree it's a bit weird. The goal is to be able to mock the `createRepository` function while being able to access the underlying `orasStore` struct fields. Since the `orasStore` is not an interface, the best way I found was to define the function signature as a struct member https://github.com/deislabs/ratify/blob/845686ca8fc067594043229bd2ac865e7a0e9d92/pkg/referrerstore/oras/oras.go#L97. When I tried to specify the receiver in the struct member definition here, go started complaining, which is why I removed the receiver and moved it to be a parameter. I've tried looking around and haven't had much luck on how to do this better. I can try using `reflection` but I've found that to be finicky too. Any suggestions here?

Reply:
yeah, I feel it would be too complex to test Oras without mocking up `createRepository`. I'm considering another option:
1. Define a new interface for `createRepository` which contains the `createRepository` API.
2. Embed this interface to `orasStore` so taht `orasStore` can call `createRepository` in the same way.
3. Implement the interface with the original logic.
4. For the test, we only need to mock the new inteface and replace with the mocked implementation.

Reply:
Yes! great idea. I'll try that. thanks

Reply:
added a flavor of this

Reply:
hmm... so this doesn't quite work since we still will need to pass in the `orasStore` reference as a parameter. if you embed this new `RepositoryHelpers` interface in the existing `orasStore` struct the `orasStore` struct member implementation of `createRepository` will be called every time even if a new `RepositoryHelpers` mock that implements the interface is provided as override on `orasStore` instantiation in the tests. We would need to specifically change the `orasStore` methods to call the interface `createRepository` method using something like `store.RepositoryHelpers.createRepository(...)`. When you do that, then the tests pass but regular scenarios fail since for the `store.RepositoryHelpers` hasn't been set to anything explicitly like it has in the tests.

Reply:
I see, that makes sense to me. In this case, I'm good with your original implementation, sorry for back-and-forthing.

Reply:
Thanks for the suggestions here. Unfortunate that we couldn't find a better solution to this. I'll revert back.